### PR TITLE
fix compiler warnings (printf security)

### PR DESCRIPTION
**Pull request by machinekoder**
The warnings are quite annoying on Ubuntu 14.04.

@ArcEye Can you please review this PR.

### CODE REVIEW

File: src/rtapi/rtapi_pci.h [hunk 1]
```diff
@@ -104,7 +104,6 @@ int pci_disable_device(struct pci_dev *dev);
           pci_resource_start((dev), (bar)) + 1))
 
 void __iomem *pci_ioremap_bar(struct pci_dev *pdev, int bar);
-inline void iounmap(volatile void __iomem *addr);
```
Comment (ArcEye) [inline src/rtapi/rtapi_pci.h]: What is the thought behind deleting this inline?  That is what the xenomai build fails on.

Comment (machinekoder) [inline src/rtapi/rtapi_pci.h]: This line throws a lot of warnings since iounmap is not defined and not used in other builds, except in the source file.

I should move this into another PR.

Comment (ArcEye) [inline src/rtapi/rtapi_pci.h]: The failure is line 759 of hm2_pci.c
It includes rtapi_pci.h but now does not get iounmap() defined, which it uses.

AFAIR that job has the CFLAGS set slightly differently, so it fails on various warnings that others pass.

As it is used externally by other components, probably best left as it was.

File: src/machinetalk/support/unionread.c [hunk 2]
```diff
@@ -47,6 +46,7 @@ bool print_container(pb_istream_t *stream)
     tag = taghdr >> 3;
     wiretype = taghdr & 0x07;
 #else
+    bool eof;
```
Comment (ArcEye) [inline src/machinetalk/support/unionread.c]: Presumably this is to get rid of an unused variable warning?

To my mind the whole \#if \#else construct is meaningless  as soon as `#if 1` is written.
It should be replaced with the code under \#else being in a commented section for reference if of use.

**Comment by ArcEye** [comment]
I have updated that last bit myself, because it was a handy way of giving the system a kick up the ***.
It had stalled trying to fetch some stuff

File: src/machinetalk/support/unionread.c [hunk 3]
```diff
@@ -39,19 +39,18 @@ bool print_container(pb_istream_t *stream)
...
     pb_wire_type_t wiretype;
 
-#if 1
     if (!pb_decode_varint(stream, &taghdr)) {
 	printf("Parsing taghdr failed: %s\n", PB_GET_ERROR(stream));
     }
     tag = taghdr >> 3;
     wiretype = taghdr & 0x07;
-#else
+/**********************************************************
```
Comment (machinekoder) [inline src/machinetalk/support/unionread.c]: I have no idea if it is any good and I don't think to put it into a comment makes it any better. Can we just remove it from the code? If anyone really needs this piece of code he can find it in the git history.

Comment (ArcEye) [inline src/machinetalk/support/unionread.c]: I think it was just @mhaberler's way of showing an alternative method.
He likes \#if 0 as a way of commenting out :smile:

Lets leave it in plain sight, you won't get your warning and anyone browsing will see it.
Don't know if they would think to look and see if anything had been deleted that might be useful
